metrics-operator/main.go

```diff
@@ -194,7 +194,6 @@ func main() {
 			Scheme:                mgr.GetScheme(),
 			Log:                   analysisLogger.V(env.AnalysisControllerLogLevel),
 			MaxWorkers:            2,
-			Namespace:             env.PodNamespace,
```

Comment:
this changes the original behaviour tho... We had before KLT namespace as default, now only the same as the analysis...

Reply:
But was there a requirement that we have the KLT namespace as default one ? I assume this was a bug also according to this ticket https://github.com/keptn/lifecycle-toolkit/issues/2099

Reply:
yes just read that 😸 ok than I guess it works as intended 📗

Reply:
we may consider using the KLT namespace as a fallback, if we do not find anything (although I'm not sure if this is the best solution) - https://github.com/keptn/lifecycle-toolkit/blob/a04803625526b48b1492f344a5641a1603ae3d4d/lifecycle-operator/controllers/common/helperfunctions.go#L94 
However, I would, if not explicitly set, definitely use the same namespace as the analysis as a default, instead of the KLT namespace, as that's the usual pattern when it comes to referencing other objects (e.g. pod -> configMap)

Reply:
We can introduce multiple fallbacks.

If namespace set -> search in this particular (in case of notFound -> fail)

If namespace not set -> search in parent namespace (in this case analysis), if not found -> search KLT default namespace


Would this work for you guys ?